### PR TITLE
[DEV-6622] COVID Tooltip on Agency V2

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**

Agency V2 Tooltip when COVID spending is present, tooltip also links to agency page with active section.

**Technical details:**

See commits!

**JIRA Ticket:**
[DEV-6622](https://federal-spending-transparency.atlassian.net/browse/DEV-6622)

**Mockup:**
https://bahdigital.invisionapp.com/share/WUIAG3CQ4JG#/screens/296054001_DEFC

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [x] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [ ] Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
- [x] Verified mobile/tablet/desktop/monitor responsiveness
`N/A` Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
`N/A` [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [x] Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [x] Code review complete
- [ ] DTUI Release

### CODE REVIEW

**Review by brett-varney — CHANGES_REQUESTED** [review]
At some width/name combos, tip is off the page:
image

**Comment by brett-varney** [comment]
Tip disappears as soon as pointer is moved from badge, which is very little space to get to DEFC link. Maybe a short delay before it disappears?

File: src/js/components/covid19/recipient/map/MapWrapper.jsx [hunk 1]
```diff
@@ -112,11 +116,8 @@ export default class MapWrapper extends React.Component {
 
     mapReady = () => {
         // map has mounted, load the state shapes
-        this.setState({
-            mapReady: true
-        }, () => {
-            this.prepareMap();
-        });
+        this.props.onMapLoaded(true);
+        this.prepareMap();
```
Comment (brett-varney) [inline src/js/components/covid19/recipient/map/MapWrapper.jsx:120]: just a question without digging deeper: should these lines be reversed? will there be any timing or logic problems if onMapLoaded finishes before prepareMap starts?

**Comment by brett-varney** [comment]
Getting a warning when clicking the DEFC link:
Warning: Failed prop type: The prop `isMapLoaded` is marked as required in `MapWrapper`, but its value is `undefined`.
    in MapWrapper (created by MapContainer)
    in MapContainer (created by ConnectFunction)

**Comment by brett-varney** [comment]
Clicking DEFC link goes to spending by sub-agency section of covid page, but not to the actual table page the agency is on, so user must click through or search again. Is that adequate? expected behavior not clear in AC

**Comment by brett-varney** [comment]
Page-breaking error when using browser back button to return to agency page from COVID page:

Uncaught ReferenceError: timeout is not defined
    Covid19Container webpack:///./js/containers/covid19/Covid19Container.jsx?:155
    React 5
        callCallback
        invokeGuardedCallbackDev
        invokeGuardedCallback
        safelyCallDestroy
        commitUnmount

**Comment by brett-varney** [comment]
@MaxwellKendall I found a few things worth checking, and it has merge conflicts, but considering how much bigger a change this turned out to be than the ticket implies, really good work!

**Review by brett-varney — COMMENTED** [review]
comment

**Comment by MaxwellKendall** [comment]
> At some width/name combos, tip is off the page:
> image

This is fixed with the change sto the component library which this depends on!

File: src/js/components/covid19/recipient/map/MapWrapper.jsx (hunk 1 above)
Comment (MaxwellKendall) [reply src/js/components/covid19/recipient/map/MapWrapper.jsx]: Thanks for asking this question. There were some pieces missing here. I added a condition to `componentDidUpdate` to replicate the behavior above.

**Comment by MaxwellKendall** [comment]
> Page-breaking error when using browser back button to return to agency page from COVID page:

☝️ should be fixed


 > Clicking DEFC link goes to spending by sub-agency section of covid page, but not to the actual table page the agency is on
 
 
 ☝️  this is outside the scope of the A/C
 
 
 > Failed prop type: The prop isMapLoaded

I think this should be fixed as well but my machine is blowing up whenever I start our dev server.

**Comment by brett-varney** [comment]
> 
> 
> @brett-varney
> 
> > Page-breaking error when using browser back button to return to agency page from COVID page:
> 
> ☝️ should be fixed
> 
> > Clicking DEFC link goes to spending by sub-agency section of covid page, but not to the actual table page the agency is on
> 
> ☝️ this is outside the scope of the A/C
> 
> > Failed prop type: The prop isMapLoaded
> 
> I think this should be fixed as well but my machine is blowing up whenever I start our dev server.

These are fine now; I guess we just need to make sure the tip stays fully on the page.

**Comment by MaxwellKendall** [comment]
@brett-varney that is handled by the update the component library.

**Review by brett-varney — APPROVED** [review]

**Pull request merged**